### PR TITLE
- Fix under linking, removed unnecessary header file list to be installed

### PR DESCRIPTION
**Pull request by atanu-mandal**
- Fixed under linking
- Removed the unnecessary header file list to be installed
- This is pure run time loadable, .a is not necessary, removed static flag from configure.ac
- Added #ifdef __cplusplus in sai_vm_event_log.h
- Cleaned up files

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
-AM_INIT_AUTOMAKE([subdir-objects])
+AM_INIT_AUTOMAKE([subdir-objects foreign])
 AC_CONFIG_SRCDIR([src])
 AC_CHECK_HEADERS([libxml/parser.h])
 
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
+AC_DISABLE_STATIC
 LT_INIT([shared])
```
Comment (johntconklin) [inline configure.ac]: LT_INIT([shared]) should already do the same thing as AC_DISABLE_STATIC.  It's not clear to me why it's needed.

File: opx/sai_vm_event_log.h [hunk 2]
```diff
@@ -18,9 +18,16 @@
...
+#ifdef __cplusplus
+extern "C"
+{
+#endif
 #include "event_log.h"
 #include "event_log_types.h"
 #include "sai_event_log.h"
+#ifdef __cplusplus
+}
+#endif
```
Comment (johntconklin) [inline opx/sai_vm_event_log.h]: Adding extern "C" directives around #include directives seems very wrong.  Each header file should have extern "C" directives around their *own* public function/variable declarations

File: Makefile.am [hunk 3]
```diff
@@ -30,6 +24,6 @@ src/qos/sai_vm_queue.c         src/sai-db/sai_db_gen_utils.cpp \
 src/qos/sai_vm_sched_group.c   src/sai-db/sai_db_init.cpp \
 src/sai_vm_npu_api_query.c  src/sai_vm_npu_init_config.c  src/sai_vm_shell.c src/tunnel/sai_vm_tunnel.c
 
-libsai_npu_la_LDFLAGS = -lopx_logging -lopx_common -lsai-common-utils -lopx_db_sql
+libsai_npu_la_LDFLAGS = -zdefs -lopx_logging -lopx_common -lsai-common-utils -lopx_db_sql
```
Comment (johntconklin) [inline Makefile.am]: Hmm, -zdefs does essentially the same thing as dpkg-shlibdeps's underlinking checks.  There's no harm in having it here, but it's worth considering turning this on for all the opx repositories.

File: configure.ac (hunk 1 above)
Comment (atanu-mandal) [inline configure.ac]: Without AC_DISABLE_STATIC libsai-npu.a is generated which we do not want.

File: opx/sai_vm_event_log.h (hunk 2 above)
Comment (atanu-mandal) [inline opx/sai_vm_event_log.h]: This is not required. 
https://github.com/open-switch/opx-sai-vm/pull/2/commits/fc4c03a5e334f2b1a14d1aff6c6eeac95cb6328f

**Pull request merged**